src/core/meshcop/border_agent_ephemeral_key.cpp

```diff
@@ -397,6 +402,35 @@ const char *EphemeralKeyManager::StateToString(State aState)
     return kStateStrings[aState];
 }
 
+Error EphemeralKeyManager::GenerateTapAndKeyStart(char *aTap, uint32_t aTimeout, uint16_t aUdpPort)
+{
+    Error    error = kErrorNone;
+    uint32_t randomResult;
+    char     verhoeffChecksum;
+    int  length = 0;
+
+    VerifyOrExit(aTap != nullptr, error = kErrorInvalidArgs);
+
+    // Generate 8-digit random number
+    randomResult = Random::NonCrypto::GetUint32();
```

Comment:
Here we can use a random number from the crypto generator:

```suggestion
    VerifyOrExit(Random::Crypto::FillBuffer(reinterpret_cast<uint8_t *>(&randomResult), sizeof(randomResult)) == kErrorNone, error = kErrorFailed);
```

Reply:
done